agent/agent_test.go

```diff
@@ -117,7 +126,10 @@ func TestRunnerRun(t *testing.T) {
 	t.Run("should record metrics", func(t *testing.T) {
 		mSink := new(mockPassthroughSink)
 		sinkFactory := registry.NewSinkFactory()
-		sinkFactory.Register("mock-sink", newMockSinkFn(mSink))
+		err := sinkFactory.Register("mock-sink", newMockSinkFn(mSink))
```

Comment:
@scortier all such pattern can combined like this. 
if err := sinkFactory.Register("mock-sink", newMockSinkFn(mSink)) err != nil {
t.Error(err.Error())
}